starlite/config/logging.py

```diff
@@ -253,7 +252,7 @@ class StructLoggingConfig(BaseLoggingConfig, BaseModel):
         - requires 'structlog' to be installed.
     """
 
-    processors: Optional[Iterable[Processor]] = Field(default_factory=default_structlog_processors)  # pyright: ignore
+    processors: Optional[List[Processor]] = Field(default_factory=default_structlog_processors)  # pyright: ignore
```

Comment:
It feels like we can be a little bit more generic here and use `Collection` instead.

However, I believe that the issue itself should be reported to structlog (yet we should adapt Starlite logging config to protect users from unexpected behaviour)